spring-integration-test/src/main/java/org/springframework/integration/test/context/MockIntegrationContext.java

```diff
@@ -111,7 +111,7 @@ public void afterSingletonsInstantiated() {
 	}
 
 	List<AbstractEndpoint> getAutoStartupCandidates() {
-		return this.autoStartupCandidates;
+		return new ArrayList<>(this.autoStartupCandidates);
```

Comment:
I think I'm OK with what you are suggesting, but this has to be like `Collections.unmodifiableList()` instead.
We indeed do like that in many other places where we return a collection from the object.
I missed this here because it is not public API, but as we see turns out it does not matter if we may deal with concurrent modifications. 😄

Reply:
ok, I was just not returning unmodifiableList to not break any user, who is in fact modifying. but you're right, it's not public anyways